tsdb/store.go

```diff
@@ -373,41 +373,64 @@ func (s *Store) ShardIteratorCreator(id uint64) influxql.IteratorCreator {
 	return &shardIteratorCreator{sh: sh}
 }
 
+// shardExecuteFn checks if a function should be run on a shard and returns
+// a function to execute on that shard if it should.
+type shardExecuteFn func(shardID uint64, sh *Shard) func() error
```

Comment:
I don't understand.  Is it to avoid starting a goroutine for a shard that might not need to be processed?  If so, I'd rather keep it simple and kick off `len(s.shard)` goroutines and have the func exit early if it's filter does not match the shard.